src/error.rs

```diff
@@ -53,6 +54,8 @@ impl<'tcx> Error for EvalError<'tcx> {
         match *self {
             EvalError::FunctionPointerTyMismatch(..) =>
                 "tried to call a function through a function pointer of a different type",
+            EvalError::ZstPointerDeref =>
+                "tried to deref a pointer to the ZST allocation",
```

Comment:
@oli-obk ZST writes should be allowed to be called, and the `Memory` code should simply ignore it. This is how it was already working - what caused you to need to change the alignment that started this conversation?